src/include/OSL/device_string.h

```diff
@@ -28,21 +28,42 @@ OSL_NAMESPACE_ENTER
 // stored in the table, in the 16 bytes preceding the characters.
 
 struct DeviceString {
+#if defined(__CUDA_ARCH__) && OPTIX_VERSION >= 70000
+    size_t      m_chars;
```

Comment:
Let's put a comment here:   `// For OptiX 7+, the ustring is represented as the hash`

Actually...

The whole comment above that documents DeviceString is incorrect. Can you rework this to correctly describe it how it works for OptiX 6 vs 7?

Reply:
Ah yes, will do!